src/sentry/snuba/metrics_layer/query.py

```diff
@@ -81,76 +81,109 @@ def run_query(request: Request) -> Mapping[str, Any]:
         raise e
 
     # If we normalized the start/end, return those values in the response so the caller is aware
-    results = {**snuba_results, "modified_start": start, "modified_end": end}
+    results = {
+        **snuba_results,
+        "modified_start": start,
+        "modified_end": end,
+        "indexer_mappings": mappings,
+    }
     metrics.incr(
         "metrics_layer.query",
         tags={"referrer": request.tenant_ids["referrer"] or "unknown", "status": "success"},
     )
     return results
 
 
-def resolve_metrics_query(metrics_query: MetricsQuery) -> MetricsQuery:
+RELEASE_HEALTH_ENTITIES = {
+    "c": EntityKey.MetricsCounters,
+    "d": EntityKey.MetricsDistributions,
+    "s": EntityKey.MetricsSets,
+}
+
+GENERIC_ENTITIES = {
+    "c": EntityKey.GenericMetricsCounters,
+    "d": EntityKey.GenericMetricsDistributions,
+    "s": EntityKey.GenericMetricsSets,
+}
+
+
+def _resolve_metrics_entity(mri: str) -> EntityKey:
+    parsed_mri = parse_mri(mri)
+    if parsed_mri is None:
+        raise ValueError(f"{mri} is not a valid MRI")
+
+    if parsed_mri.namespace == "sessions":
+        return RELEASE_HEALTH_ENTITIES[parsed_mri.entity]
+
+    return GENERIC_ENTITIES[parsed_mri.entity]
+
+
+def resolve_metrics_query(
+    metrics_query: MetricsQuery,
+) -> tuple[MetricsQuery, Mapping[str : str | int]]:
```

Comment:
That seemed heavyweight, and I think it's because all these functions are meant to be internal. So I didn't change the signature but I did mark all the functions as private so its clear they are only meant to be used by this module.